pkg/auth/service.go

```diff
@@ -1483,6 +1496,18 @@ func (a *APIAuthService) WritePolicy(ctx context.Context, policy *model.Policy)
 	}
 	createdAt := policy.CreatedAt.Unix()
 
+	if overwrite { // Update existing policy
```

Comment:
Think we need to write to create and update in case we fail - as update API should only update existing one.